### PR TITLE
[dhcp_server_test] Add multiple vlans test and config change test

### PR DESCRIPTION
**Pull request by w1nda**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

### CODE REVIEW

File: tests/dhcp_server/test_dhcp_server.py [hunk 1]
```diff
@@ -738,6 +389,265 @@ def test_dhcp_server_port_based_assigenment_single_ip_mac_swap(
             test_xid=test_xid,
             dhcp_interface=vlan_name,
             expected_assigned_ip=expected_assigned_ip_0,
-            exp_server_ip=gateway,
+            exp_gateway=gateway,
             net_mask=net_mask
         )
+
+
+@pytest.mark.parametrize("option_info", [["string", "#hello i'm dhcp_server!"]])
```
Comment (yaqiangz) [inline tests/dhcp_server/test_dhcp_server.py:397]: better test comma string

File: tests/dhcp_server/test_dhcp_server.py [hunk 2]
```diff
@@ -738,6 +389,265 @@ def test_dhcp_server_port_based_assigenment_single_ip_mac_swap(
...
+    ptfadapter,
+    parse_vlan_setting_from_running_config,
+    option_info
+):
+    """
+        Test dhcp server packets if carry the customized options as expected
+    """
+    test_xid = 8
+    vlan_name, gateway, net_mask, vlan_hosts, vlan_members_with_ptf_idx = parse_vlan_setting_from_running_config
+    expected_assigned_ip = random.choice(vlan_hosts)
```
Comment (yaqiangz) [inline tests/dhcp_server/test_dhcp_server.py:410]: Better to log info random result

File: tests/dhcp_server/test_dhcp_server.py (hunk 1 above)
Comment (w1nda) [reply tests/dhcp_server/test_dhcp_server.py]: updated

File: tests/dhcp_server/test_dhcp_server.py (hunk 2 above)
Comment (w1nda) [reply tests/dhcp_server/test_dhcp_server.py:432]: updated

File: tests/dhcp_server/test_dhcp_server_multi_vlans.py [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+        + [add_vlan_ip_patch(vlan_name, ip)[0] for ip in vlan_info['interface_ipv4']] \
+        + [add_vlan_ip_patch(vlan_name, ip)[0] for ip in vlan_info['interface_ipv6']] \
+        + [add_vlan_member_patch(vlan_name, member)[0] for member in vlan_info['members']]
+
+    # split single vlan into two vlans
+    vlan_prefix = vlan_info['interface_ipv4'][0]
+    vlan_net = ipaddress.ip_network(address=vlan_prefix, strict=False)
+    vlan_nets = list(vlan_net.subnets(prefixlen_diff=2))
+    member_count = len(vlan_member_with_ptf_idx)//4
+    for i in range(3):
```
Comment (w1nda) [inline tests/dhcp_server/test_dhcp_server_multi_vlans.py:75]: range(3)?

File: tests/dhcp_server/test_dhcp_server_multi_vlans.py [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+            smaller_subnet = list(vlan_nets[i].subnets(new_prefix=smaller_prefix_length))[0]
+            four_vlans_info.append(
+                {
+                    'vlan_name': 'Vlan40%s' % i,
+                    'vlan_gateway': str(list(smaller_subnet.hosts())[0]),
+                    'interface_ipv4': str(list(smaller_subnet.hosts())[0]) + '/' + str(smaller_prefix_length),
+                    'vlan_subnet_mask': str(smaller_subnet.netmask),
+                    'vlan_hosts': [str(host) for host in list(smaller_subnet.hosts())[1:]],
+                    'members_with_ptf_idx': [(member, ptf_idx) for member, ptf_idx
+                                             in vlan_member_with_ptf_idx[member_count*i:member_count*i+1]]
```
Comment (w1nda) [inline tests/dhcp_server/test_dhcp_server_multi_vlans.py:88]: Shuffle this list vlan_member_with_ptf_idx

**Review by yaqiangz — APPROVED** [review]
LGTM

**Review by wangxin — APPROVED** [review]

**Pull request merged**